src/main.rs

```diff
@@ -0,0 +1,204 @@
+#[macro_use]
+extern crate serde;
+extern crate tokio;
+
+use chrono::DateTime;
+use chrono::Local;
+use chrono::SubsecRound;
+use chrono::TimeDelta;
+use serde_json::Deserializer;
+use serde_json::Value;
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use tokio::task::JoinSet;
+use tokio::time::sleep;
+
+#[derive(Debug, Clone)]
+pub enum Instruction {
+    Lighting(Lights),
+    Pumping(Pumps),
```

Comment:
I knew it was dubious, but I chose action words for the variants of `Instructions`...with the idea to have functional groups for _future_ devices, so maybe something like:
`+`    Heating ...
`+`    Sensing
`+`    Fertilizing
`+`    Venting
`+`    Regulating ...

Other ideas I had would be:
- to categorize by devices 
- or just have one big instruction enum or struct.

To workshop the name more.. I think it may have to do with our JSON schedule, and maybe config.
Besides something more like defining instruction names based on input (with macros I'd assume), is it worth considering now which should be more readable? I'd suggest that the JSON might want to be more readable, because users will be handling those files. OTOH displaying back to user from this program will be rust based.

right now from the running rust program I can see...
`light schedule : [Lighting(LightsOn(2024-11-18T05:05:38Z)), Lighting(LightsOff(2024-11-18T05:05:49Z))]`

Where `LightsOn` is really more like `AllLightsOn` because there is only one lighting device right now.

Reply:
I like the idea of having instructions as part of a schedule. For example, we might want to issue an instruction to something that isn't necessarily a `device` - and keeping things as `instructions` liberates us from the need to have knowledge of what a `device` is. So I think you're on the right track there!

In my head the instruction syntax parses as:
`Instruction (issued at a particular time) -> Functional Group -> Action`

So for lighting, it would look like: 
`Instruction -> Lighting -> Off`

I think there's a way we can (potentially with a plugin system, potentially with something different) group _things_ into categories that make sense, while still maintaining the ability to write a single, simple schedule or other such automations.